src/lib/user.rs

```diff
@@ -35,25 +35,51 @@ impl User {
             correct: 0,
             wrong: 0,
             ranking: Ranking::Noob,
-            score: 0,
+            points: 0,
         }
     }
 
     /// Called when a user answers correctly
     /// qta = (Q)uestion (T)ype (A)nswered - Can be Easy, Medium, or Hard
     /// Depending on the level of difficulty of their question, this will determine how much to
     /// increment their score by.
-    pub fn increment(qta: Level) {
-        //TODO: Current implementation is basic and not correct, see above documentation
+    pub fn increment(&mut self, qta: &Level, incr: bool) {
+        self.points += match qta {
+            Level::Easy => {
+                if incr {
+                    5
+                } else {
+                    -5
+                }
+            }
+            Level::Medium => {
+                if incr {
+                    10
+                } else {
+                    -10
+                }
+            }
+            Level::Hard => {
+                if incr {
+                    15
+                } else {
+                    -15
+                }
+            }
+        }
     }
-
+    /**
+     * Get percentage total win of correct scores
+     * @param self: User
+     * @returns final pct: f32
+     */
     pub fn pct(&self) -> f32 {
         let total = (self.correct + self.wrong) as f32;
         if total == 0.0 {
             // If total is 0, compiler will see it as NaN
-            0.0
+            return 0.0;
```

Comment:
If you replace `return 0.0` with just `0.0` and leave off the `;` that is the same as a return statement. Also this should clear the clippy error you're getting.

Reply:
Thanks mate! I made the fix and pushed.